### PR TITLE
Truncate keys to u16::MAX in term hashmap

### PR DESCRIPTION
**Pull request by PSeitz**
Truncate keys to u16::MAX, instead e.g. storing 0 bytes for keys with length u16::MAX + 1

The term hashmap has a hidden API contract to only accept terms with lenght up u16::MAX.

### CODE REVIEW

File: stacker/src/shared_arena_hashmap.rs [hunk 1]
```diff
@@ -308,6 +312,8 @@ impl SharedArenaHashMap {
         if self.is_saturated() {
             self.resize();
         }
+        // Limit the key size to u16::MAX
+        let key = &key[..std::cmp::min(key.len(), u16::MAX as usize)];
```
Comment (adamreichold) [inline stacker/src/shared_arena_hashmap.rs:314]: There is [`impl From<u16> for usize`](https://doc.rust-lang.org/stable/std/primitive.usize.html#impl-From%3Cu16%3E-for-usize), i.e. the cast could be replaced by `.into()`.

Comment (PSeitz) [reply stacker/src/shared_arena_hashmap.rs:314]: I prefer to keep it explicit

Comment (adamreichold) [reply stacker/src/shared_arena_hashmap.rs:314]: You can explicitly write `usize::from(u16::MAX)` if you prefer while still avoiding the potentially truncating cast.

**Review by fulmicoton — APPROVED** [review]

File: stacker/src/shared_arena_hashmap.rs (hunk 1 above)
Comment (PSeitz) [reply stacker/src/shared_arena_hashmap.rs:314]: I wish rust would just coerce types that can be safely cast

here I don't think we gain much for proof reading

**Pull request merged**